src/components/landingPage/index.js

```diff
@@ -0,0 +1,270 @@
+import Footer from '../common/footer/Footer';
+import ArticleCard from './articleCard';
+import LandingPageLinks from './landingPageLinks';
+import InterestArticlesAreaUpperGreenLine from './interestArticleAreaUpperLine';
+import PopularArticleAreaUpperLine from './interestArticleAreaUpperLine/popularArticleAreaUpperHeader';
+import LandingPageSearchArea from './landingPageSearchArea';
+import LandingPageViewMoreArticles from './landingPageViewMoreArticles';
+import LandingPageArrowArea from './landingPageArrowArea';
+import displayArticles from '../../redux/actions/landingPage';
+import displayPopularArticles from '../../redux/actions/landingPage/popularArticles';
+import NavBar from '../common/navbar/navbar';
+import * as actionTypes from '../../redux/actions/actionTypes';
+
+export class LandingPage extends Component {
+  state = {
+    articles: [],
+    popularArticles: [],
+    initialLimits: {
+      limitArticlesNumber: 3,
+      limitPopularArticlesNumber: 3,
+    },
+    limitArticlesNumber: 3,
+    limitPopularArticlesNumber: 3,
+    loading: false,
+  };
+
+  componentDidMount() {
+    this.setState((prevState) => ({ ...prevState, loading: true }));
+    const { props } = this;
+    props.displayArticles().then((res) => {
+      if (res.payload.status !== 200) {
+        this.setState((prevState) => ({ ...prevState, loading: true }));
+      }
+      this.setState((prevState) => ({ ...prevState, loading: false }));
+    });
+    props.displayPopularArticles().then((res) => {
+      if (res.payload.status !== 'success') {
+        this.setState((prevState) => ({ ...prevState, loading: true }));
+      }
+      this.setState((prevState) => ({ ...prevState, loading: false }));
+    });
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps) {
+    const { data, popularArticles } = nextProps;
+    this.setState((prevState) => ({ ...prevState, articles: data, popularArticles }));
+  }
+
+  viewMoreArticles = (viewMoreArticlesButton, articlesNumber, number = 5) => {
+    const { state } = this;
+    const oldNumber = state[viewMoreArticlesButton];
+    const initialLimit = state.initialLimits[viewMoreArticlesButton];
+    const newNumber = oldNumber >= articlesNumber ? initialLimit : oldNumber + number;
+    this.setState((prevState) => ({
+      ...prevState,
+      [viewMoreArticlesButton]: newNumber,
+    }));
+  };
+
+  render() {
+    const {
+      articles,
+      popularArticles,
+      limitArticlesNumber,
+      limitPopularArticlesNumber,
+      loading,
+    } = this.state;
+    const mostTwoRecentArticles = articles.slice(0, 2);
+    const secondAndThirdArticles = articles.slice(2, 4);
+    const remainingArticles = articles.slice(4);
+    const displayFirstAndSecond = mostTwoRecentArticles.map((article) => (
+      <a
+        href={`${actionTypes.BACKEND_URL}/api/${actionTypes.VERSION}/articles/${article
+          && article.slug}`}
+      >
+        <ArticleCard
+          key={article && article.slug}
```

Comment:
Could you explain for me the logic behind `article && article.slug` etc...

Reply:
Due to the fact that render lifecycle renders before componentDidMount. We have to wait for the lifecycle to render. So this is to say, if article is rendered, then use article.slug.

Reply:
Cool